shell/platform/embedder/embedder_semantics_update.cc

```diff
@@ -175,4 +193,61 @@ void EmbedderSemanticsUpdate2::AddAction(
   });
 }
 
+std::pair<size_t, const FlutterStringAttribute**>
+EmbedderSemanticsUpdate2::CreateStringAttributes(
+    const StringAttributes& attributes) {
+  // Minimize allocations if attributes are empty.
+  if (attributes.empty()) {
+    return std::make_pair(0, nullptr);
+  }
+
+  auto result = std::make_unique<std::vector<const FlutterStringAttribute*>>();
+  result->reserve(attributes.size());
```

Comment:
I might be tempted to add just one comment to this method above this declaration that this is the list of embedder APIs FlutterStringAttributes mapping to the input list of internal StringAttribute objects, and whose innards we'll be returning from this method.

Reply:
Added this comment. I also included details from your other comment suggestion above: https://github.com/flutter/engine/pull/44616#discussion_r1297825808